File: Spipama.API/SpipamaClientPublicWebApp/src/app/@core/viewModels/ActionPlanViewModel.ts
import { DatePipe, DecimalPipe } from "@angular/common";

export class ActionPlanViewModel {
  id: string;
  name: string;
  strategyName: string;
  startYear: number;
  endYear: number;
  activeYear: number;
  status: number;
  budgetTotal: DecimalPipe;
  budgetCapital: DecimalPipe;
  budgetCost: DecimalPipe;
  fileRef: string;
  fileName: string;
  createdBy: string;
  createdOn: DatePipe;
  updatedBy: string;
  updatedOn: DatePipe;
  isDeleted: boolean;


  constructor(
    id: string = null,
    name: string = null,
    strategyName: string = null,
    startYear: number = null,
    endYear: number = null,
    activeYear: number = null,
    status: number = null,
    budgetTotal: DecimalPipe = null,
    budgetCapital: DecimalPipe = null,
    budgetCost: DecimalPipe = null,
    fileRef: string = null,
    fileName: string = null,
    createdBy: string = null,
    createdOn: DatePipe = null,
    updatedBy: string = null,
    updatedOn: DatePipe = null,
    isDeleted: boolean = null,
  ) {
    this.id = id;
    this.name = name;
    this.strategyName = strategyName;
    this.startYear = startYear;
    this.endYear = endYear;
    this.activeYear = activeYear;
    this.status = status;
    this.budgetTotal = budgetTotal;
    this.budgetCapital = budgetCapital;
    this.budgetCost = budgetCost;
    this.fileRef = fileRef;
    this.fileName = fileName;
    this.createdBy = createdBy;
    this.createdOn = createdOn;
    this.updatedBy = updatedBy;
    this.updatedOn = updatedOn;
    this.isDeleted = isDeleted;
  }
}

File: Spipama.Application/ViewModels/ActionPlanViewModel.cs
using Microsoft.AspNetCore.Http;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Spipama.Application.ViewModels
{
    public class ActionPlanViewModel
    {
        public Guid Id { get; set; }
        [Required]
        public string Name { get; set; }
        [Required]
        public string StrategyName { get; set; }
        [Required]
        public int StartYear { get; set; }
        [Required]
        public int EndYear { get; set; }
        [Required]
        public int ActiveYear { get; set; }
        [Required]
        public int Status { get; set; }
        [Required]
        public decimal? BudgetTotal { get; set; }
        [Required]
        public decimal? BudgetCapital { get; set; }
        [Required]
        public decimal? BudgetCost { get; set; }
        public DateTime createdOn { get; set; }
        public string FileRef { get; set; }
        public string FileName { get; set; }
        public string IndicatorFileRef { get; set; }
        public string IndicatorFileName { get; set; }
        public bool IsDeleted { get; set; }
    }
}
